server/src/controller_test.py

```diff
@@ -89,6 +110,81 @@ def json(self):
         'ok': True
     }
 
+######## AUTH TESTS ###########
+
+def test_authenticate_user():
+    mockUser = mocks.MockUser("testuser", "testpassword")
+    mockRequest = {"username": "testuser", "password": "testpassword"}
+    mockWrongRequest = {"username": "testuser"}
+    correct = controller.authenticate_user(mockRequest, mocks.MockFilterProvider(mockUser))
+    incorrect = controller.authenticate_user(mockWrongRequest, mocks.MockFilterProvider(mockUser))
+    data = json.loads(correct.body)
+    assert data['ok'] == True
+    data = json.loads(incorrect.body)
+    assert data['ok'] == False
+
+
+######## ADD TESTS #############
+
+def test_add_user(monkeypatch):
+    class SlackUsersMock:
+        def __init__(*args, **kwargs):
+            return
+        def json(self):
+            return {
+                'ok': True,
+                'members': [
+                    {
+                        'deleted': False,
+                        'name': 'snax',
+                        'profile': {
+                            'real_name_normalized': 'Hot Dog',
+                            'image_192': 'https://fakeurl.coolbeans'
+                        }
+                    },
+                    {
+                        'deleted': False,
+                        'name': 'fail',
+                        'profile': {
+                            'real_name_normalized': 'I should throw',
+                            'image_192': 'https://fakeurl.coolbeans'
+                        }
+                    }
+                ]
+            }
+    monkeypatch.setattr(requests, 'get', SlackUsersMock)
+    response = controller.get_slack_users()
+    assert type(response) is HTTPResponse
```

Comment:
isinstance(response, HTTPResponse) is better in general because it will return true if the type is a child of the one you're checking against